### ISSUE TITLE
QR URL for easyAPI

### ISSUE THREAD

**Issue by scropion86** [open-wa/wa-automate-nodejs#3016]
### Are you using the latest version of the library?

- [X] I have checked and am using the latest version of the library.

### What type of session are you experiencing this issue on?

Multi-device and I have set multiDevice to true in my config OR am using the --multi-device flag

### What type of host account are you experiencing this issue on?

Personal account (normal)

### Mode

EASY API/CLI

### Current Behavior

When enabling `--ezqr --popup true --ev 'https://webhook.site/7dce32e9…..`

the webhock received everything at launch except the QR URL such this message` “Scan the qr code at https://qr.openwa.cloud/1681d”`

and since I enabled popup flag I was expecting to be able to see the QR from `“server:port/qr"` but because I am using API key o got “{"error":"unauthorised"}”

### Expected Behavior

- should receive the QR URL in webhook launch events
- should be able to access `“server:port/qr"` without API Key

### Steps To Reproduce

npx @open-wa/wa-automate -p 1234 --ezqr --popup true --ev 'https://webhook.site/7dce32e9-561d-4.........'

### create() code OR full CLI command + CONFIG

```markdown
npx @open-wa/wa-automate -p 1234 --ezqr --popup true --ev 'https://webhook.site/7dce32e9-561d-4.........'
```

### DEBUG INFO

```markdown
┌─────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│       (index)       │                                                                   Values                                                                   │
├─────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│     WA_VERSION      │                                                                 '2.2301.6'                                                                 │
│       PAGE_UA       │ 'WhatsApp/2.2147.16 Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36' │
│ WA_AUTOMATE_VERSION │                                                                  '4.55.0'                                                                  │
│   BROWSER_VERSION   │                                                       'HeadlessChrome/109.0.5412.0'                                                        │
│         OS          │                                                                'Linux 5.15'                                                                │
│      START_TS       │                                                               1673879684176                                                                │
│      RAM_INFO       │                                                      'Total: 8.20 GB | Free: 7.25 GB'                                                      │
│    PPTR_VERSION     │                                                                  '19.5.2'                                                                  │
└─────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

### Environment

```markdown
- OS: 
- Node: 
- npm:
```

**Comment by smashah** [comment]
Hmmm, it's assumed that you want to block public access to the qr code if you set the api key.

Maybe there can be an environment variable to bypass any authentication specifically on the popup?

**Comment by fire17** [comment]
watching :)
+ can anyone share a link to docs where this is officially covered ? 

i am spinning up open-wa with npx on a python subprocess, i can create a new directory, open the process from there,
and get a fresh wa api ready for qr scanning... I wish to know how to export it and also get the changing qr if it wasn't scanned it time. I assume this is what the webhook option is for.. Some guidance or material would help :) thanks alot

**Comment by scropion86** [comment]
Yes this might be a smart workaround , environment variable or CLI flag in
case of EasyApi

**Comment by scropion86** [comment]
> watching :)
> 
> * can anyone share a link to docs where this is officially covered ?
> 
> i am spinning up open-wa with npx on a python subprocess, i can create a new directory, open the process from there, and get a fresh wa api ready for qr scanning... I wish to know how to export it and also get the changing qr if it wasn't scanned it time. I assume this is what the webhook option is for.. Some guidance or material would help :) thanks alot

if you mean documentation for API here is the Link
https://openwa.dev/docs/configuration/command-line-options

**Comment by smashah** [comment]
something is missing here, how are you getting `{"error":"unauthorised"}` if:

- you don't have `-k` in your command (are you using `cli.config.json`?)
- that authentication layer is only available once the api is up and the session is running?

Can you show me a screen recording (including terminal) of you experiencing this problem please + `-v` logs

**Comment by smashah** [comment]
Also @scropion86 @fire17  you can constantly refresh that `--ezqr` image as it is not cached and always returns the latest qr code (albeit with a delay of upload).

**Comment by fire17** [comment]
Thanks @scropion86 for the cli docs link, i've learned of a few more things that are possible
even though it didnt mention the qr there,
since im using the api with python, I usually go by https://openwa.dev/docs/api/classes/api_Client.Client
but again no mention there of onQR event

@smashah could you please advise ?
for me the best case scenario is to have an event from the client (like any other) where it fires every time a qr is prompted
This way I can update any open socket to my site with an updated qr for me to scan remotely
Please let me know if this is possible or how this could be done otherwise from my env?..
Thanks a lot 🙏 😇

**Comment by scropion86** [comment]
> something is missing here, how are you getting `{"error":"unauthorised"}` if:
> 
> * you don't have `-k` in your command (are you using `cli.config.json`?)
> * that authentication layer is only available once the api is up and the session is running?
> 
> Can you show me a screen recording (including terminal) of you experiencing this problem please + `-v` logs

sorry for confusing you , the error `{"error":"unauthorised"}` was a result from other session which -k was used.
and i am not using `cli.config.json`

**Comment by smashah** [comment]
@fire17 have you tried the python socket client?

https://github.com/open-wa/wa-automate-socket-client-python

@scropion86 ok so there should be no limitations on the popup URL. What is the remaining problem then?

**Comment by scropion86** [comment]
> 

--ezqr URL is not sent to the webhook if --ev 'https://webhook.site/7dce32e9…..

**Comment by smashah** [comment]
todo:

- [ ] qrUrl event namespace should be added to default `ef`

**Comment by smashah** [comment]
@github-actions run

<details>
<summary>⚡ Release!  ⚡</summary>

```js
(async () => {
function exec(cmd) {
try {
	execSync(cmd, {stdio: 'inherit'})
}  catch (error) {
    console.log(`Status Code: ${error.status} with '${error.message}'`);
 }
}
//set the version type
process.env.VERS = "patch"

// Config
const gitUserEmail = "github-actions[bot]@users.noreply.github.com";
const gitUserName = "github-actions[bot]";

exec(`echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > .npmrc`);
exec(`git config --global user.email "${gitUserEmail}"`);
exec(`git config --global user.name "${gitUserName}"`);
exec(`npm i -D`);
exec(`npm run release-ci $VERS`);

// types only package
exec('npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN')
exec(`npm config set //registry.npmjs.org/:_authToken $NPM_TOKEN && npm run types:publish `);
exec(`git commit -a -m 'updated types-only package'`);
exec(`git push --force`);

//comment on the issue
var result = execSync(`npx auto-changelog -o ./tempchangelog.txt --commit-limit false --template ./compact-keepachangelog.hbs --stdout`).toString();

    await postComment(result);

//create changelog image
exec(`npm run release-image`);
exec(`git commit -a -m 'updated release-image'`);
exec(`git push --force`);
  })();
```
</details>

**Comment by smashah** [comment]
# Changelog

## 🚀 Release [4.55.2](https://github.com/open-wa/wa-automate-nodejs/compare/4.55.1...4.55.2) (2023-01-21)

- 📝 docs update [`6fbd9a3`](https://github.com/open-wa/wa-automate-nodejs/commit/6fbd9a37a5311a3244195f5192c5a5204042a4d2)
- Release 4.55.2 [`df8086b`](https://github.com/open-wa/wa-automate-nodejs/commit/df8086b363db18a2b93dc7b100878ed604d9d2a9)
- 🔧 added `qrUrl` to list of default ev filter #3016 [`7d4911a`](https://github.com/open-wa/wa-automate-nodejs/commit/7d4911a24e20c0451427f942844cf6bfbe37f648)
- updated types-only package [`da397f2`](https://github.com/open-wa/wa-automate-nodejs/commit/da397f20cf659855c246bbf5039a3076cbec63b5)

**Comment by fire17** [comment]
> @fire17 have you tried the python socket client?
> 
> https://github.com/open-wa/wa-automate-socket-client-python

Yes @smashah this is exactly what I am using
I figured my issue should go there perhaps...  [Getting QR to client before connection #7](https://github.com/open-wa/wa-automate-socket-client-python/issues/7)
(hope you watch the issues there as you do here lol 😋 )
